src/modules/tasksClass.js

```diff
@@ -0,0 +1,112 @@
+      <div class="col-1">
+      <!-- task selection form-->
+      <form action="task-select-form" class="task-select-form d-flex justify-content-start align-items-center">
+      <label id="task-select" for="task-select-input"> Select a task</label>
+      <input id="task-select-input" class="form-check-input task-select-input" type="checkbox"/>
+      </form>
+      </div>
+      <!--task list-->
+      <div class="col-10">
+      <p id="task-text" class="task-text">${taskText}</p>
+      </div>
+      <!-- delete and drag btn-->
+      <button id="remove-btn" class="remove-btn bi bi-three-dots-vertical btn btn-sm col-1"></button>`;
+    displayContainer.appendChild(taskContainer);
+    this.displayAllTasks();
+  }
+
+  displayAllTasks = () => {
+    const tC = JSON.parse(localStorage.getItem('taskList'));
+    // empty container
+    const displayContainer = document.querySelector('#display-container');
+    displayContainer.innerHTML = '';
+    // begin loop
+    tC.forEach((tcTask, tcTaskIndex) => {
+      displayContainer.innerHTML += ` 
+      <div id="${tcTaskIndex}" class="task-container row px-2 ms-0 me-0">
+        <!--checkbox input col-->
+        <div class="col-1">
+        <!-- task selection form-->
+        <form action="task-select-form" class="task-select-form d-flex justify-content-start align-items-center">
+        <label id="task-select" for="task-select-input"> Select a task</label>
+        <input id="task-select-input" class="form-check-input task-select-input" type="checkbox"/>
+        </form>
+        </div>
+        <!--task list-->
+        <div class="col-10">
+        <p id="task-text" class="task-text">${tcTask.taskDescription}</p>
+        </div>
+        <!-- delete and drag btn-->
+        <button id="remove-btn" class="remove-btn bi bi-three-dots-vertical btn btn-sm col-1"></button>
+      </div>`;
+    });
+  } // showing all tasks
+
+  taskRemover = (btnIndex) => {
+    const tC = this.taskCollection.filter((task) => task.taskIndex !== btnIndex + 1);
+    // rearrange by sorting using their index
+    tC.sort((task1, task2) => task1.taskIndex - task2.taskIndex);
+    // sorting changed indexes, reassign task index by iterations
+    tC.forEach((taskItem, taskItemIndex) => {
+      taskItem.taskIndex = taskItemIndex + 1;
+    });
+    localStorage.setItem('taskList', JSON.stringify(tC));
+  }
+
+  taskEditor(elIndex, taskItemInput) {
+    const tC = this.taskCollection;
+    tC[elIndex].taskDescription = taskItemInput;
+    localStorage.setItem('taskList', JSON.stringify(this.taskCollection));
+  } // edit task
+
+  taskCompleted = (theElIndex, taskBoxValue) => {
+    const tC = this.taskCollection;
+    tC[theElIndex].taskCompletion = taskBoxValue;
+    localStorage.setItem('taskList', JSON.stringify(this.taskCollection));
+  } // task is marked complete
+
+  clearCompletedTask = (theElIndex) => {
+    const tC = this.taskCollection;
+    // tC.filter((task) => task.taskCompletion === false);
```

Comment:
As part of the [ruby best practices](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md) kindly remove this commented code from your codebase. Leaving commented code in your codebase is generally not a good practice. Commented-out code can clutter your codebase, make it harder to read, and can potentially lead to confusion or outdated information. If you need to review a previous version of your code, you can always use git history. :pray: